### DISCUSSION TITLE
Migration file in Prod and Dev are different.

### DISCUSSION THREAD

**Discussion by ghost** [prisma/prisma#21545] [Q&A]
### Question

Hi! I've been using Prisma TRPC for awhile but still a beginner.
I would like some help to clarify my unclear perplexity and save my project.

Normally, I have Prod and Dev environments.
but somehow I managed to release some of the commits to Prod without concerning the sequence of the migration file in Prod and Dev. 

The situation is in Dev migration file must be 1>2>3>4>5>6 in order
But after the release Prod has 1>2>3>5>6 (missing no.4) I already run `npx prisma migrate deploy` 

If I release again with migration file 1>2>3>4>5>6 in order from Dev to Prod what will happen?
Will it runs migration file no.4 and then Everything is fine. Or what can happen?

Thank you in advance 🙏🏼
Please save meeeeeee

**Comment by ludralph** [comment]
Hi @mindtechup 👋 

Thank you for raising this question.

When you run `npx prisma migrate deploy` in your production environment, [Prisma Migrate](https://www.prisma.io/docs/concepts/components/prisma-migrate/migrate-development-production#production-and-testing-environments) compares the applied migrations against the migration history and warns if any migrations have been modified. However, it does not issue a warning if an already applied migration is missing from the migration history.

If you release again with the full sequence of migration files (1>2>3>4>5>6) from Dev to Prod, since migrate deploy applies pending migrations, it's possible that it might apply the missing migration file no.4. However, it's important to note that migrate deploy does not detect drift (i.e., when the production database schema differs from the migration history end state - for example, due to a hotfix).

Always make sure to have backups and snapshots of your database before running migrations, especially in a production environment, to avoid data loss or corruption.

If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
